weave/flow/dataset.py

```diff
@@ -64,3 +68,18 @@ def convert_to_table(cls, rows: Any) -> weave.Table:
                     "Attempted to construct a Dataset row with an empty dict."
                 )
         return rows
+
+    @weave.op
```

Comment:
why are these ops?

Reply:
Interesting. So the root problem you are facing is that when to `weave.get` a dataset, the resulting object is a WeaveObject, and therefore do not have these methods bound. 

This feels a bit like a hack, with a few undesirable properties:
1. A bunch of calls tracked that the user probably does not care about
2. Ops that are not pure (in this case there is a side-effect). Ops are intended to be pure.
3. I think the result of this behavior is that a new version dataset/table gets created after every single method. I think this is more than desirable.

Do you agree that these are issues? Did you consider alternatives?

Reply:
RE: Properties
1. You might care about tracking row-adds to a dataset, but I think atomic unit is the "squash commit append", not per-row appends
2. I didn't know that Ops needed to be pure.  Mutations are inherently not pure, so maybe we need to rethink this.
3. Agree -- per method is noisy, but per "squash commit append" feels like it would be useful.

One alternative is to operate on the lower level `rows` object.  This feels wrong because the user is "reaching into" the Dataset object to interact with something that ostensibly should be an implementation detail, but it does immediately satisfy our constraints and `Dataset.append` effectively delegates to `Dataset.rows.append` anyways.

Another option is making something like a "baby op" which doesn't do all the versioning on op call but rather collects the mutations and only does 1 call at publish time.  That feels more correct but is also more work.

I think I prefer the 2nd option, but would not implement until we have the full mutation system working.

Reply:
I think there might be a mixing of objectives going on. 

Objective 1: Allow a "gotten" dataset to have these functions
Objective 2: Track additional parts of the mutation process.

IMO, we should just do #1. The versioning happens at the object level and we can always diff the datasets / row properties if we want to know what changed between commits